nestkernel/event.h

```diff
@@ -1053,11 +1053,13 @@ class DataSecondaryEvent : public SecondaryEvent
   {
     write_to_comm_buffer( *( supported_syn_ids_.begin() ), pos );
     write_to_comm_buffer( sender_gid_, pos );
+    DataType elem;
     for ( typename std::vector< DataType >::iterator i = coeffarray_as_d_begin_;
           i != coeffarray_as_d_end_;
           i++ )
     {
-      write_to_comm_buffer( *i, pos );
+      elem = *i;
+      write_to_comm_buffer( elem, pos );
```

Comment:
I would make this even more compact as 
```c++
const DataType elem = *i;
```
And one should probably be able to do it as a one-liner:
```c++
write_to_comm_buffer( static_cast< DataType >(*i), pos );
```
I would prefer that, keeping things as tight as possible.

I would also strongly suggest adding a comment here why this extra effort is necessary.